File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i, j = 0, len(heights) - 1
        max_area = 0

        while i < j:
            width = j - i
            curr_area = width * min(heights[i], heights[j])
            max_area = max(max_area, curr_area)

            if heights[i] < heights[j]:
                i += 1
            else:
                j -= 1

        return max_area

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0];
        int maxProfit = 0;
        int profit;

        for (const auto &price : prices) {
            minPrice = min(minPrice, price);
            profit = price - minPrice;
            maxProfit = max(maxProfit, profit);
        }
        return maxProfit;
    }
};
